BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Client/Components/MarkDownComponent.razor

```diff
@@ -0,0 +1,233 @@
+				</div>
+				<textarea class="form-control text-area-resized" @bind="markDownModel.Body"></textarea>
+			}
+			else
+			{
+				<textarea class="form-control text-area-resized" disabled @bind="markDownModel.Body"></textarea>
+			}
+		</div>
+		@if (checkPreview)
+		{
+			<div class="col-6">
+				<h3>MarkDown @MarkDownId : PreView</h3>
+				@if (!string.IsNullOrWhiteSpace(UserMail))
+				{
+					<div class="mb-3 text-right">
+						<button class="btn btn-primary btn-margin" @onclick="ReloadPreview">Reload Preview</button>
+						<button class="btn btn-primary btn-margin" @onclick="()=>CheckPreview(false)">View Html generated</button>
+						<button class="btn btn-primary btn-margin" @onclick="()=>DisplayEditor(false)">Hide Editor</button>
+					</div>
+				}
+				<div>@markupString</div>
+			</div>
+		}
+		else
+		{
+			<div class="col-6">
+				<h3>MarkDown @MarkDownId : Html Generated</h3>
+				@if (!string.IsNullOrWhiteSpace(UserMail))
+				{
+					<div class="mb-3 text-right">
+						<button class="btn btn-primary" @onclick="ReloadPreview">Reload Preview</button>
+						<button class="btn btn-primary" @onclick="()=>CheckPreview(true)">View Preview generated</button>
+					</div>
+				}
+				<div>@markDownModel.Html</div>
+			</div>
+		}
+	</div>
+}
+else
+{
+	<div class="col-6">
+		
+		@if (!displayEditor)
+		{
+			<h3>MarkDown @MarkDownId : Editor</h3>
+			<div class="mb-3 text-right">
+				<button class="btn btn-primary btn-margin" @onclick="()=>DisplayEditor(true)">Display Editor</button>
+			</div>
+		}
+		else
+		{
+			<h3>README</h3>
+		}
+		<div>@markupString</div>
+	</div>
+}
+
+@code {
+
+	[Parameter]
+	public string? UserMail { get; set; }
+
+	[Parameter]
+	public long? MarkDownId { get; set; }
+
+	private MarkupString markupString;
+
+	private string successMessage = string.Empty;
+	private string errorMessage = string.Empty;
+
+	private bool isLoading = true;
+	private bool checkPreview = true;
+	private bool displayEditor = true;
+
+	private MarkDownModel markDownModel = new MarkDownModel
+	{
+		Id = 0,
+		Body = string.Empty
+	};
+
+	protected override async Task OnInitializedAsync()
+	{
+		await base.OnInitializedAsync();
+		isLoading = true;
+		successMessage = string.Empty;
+		errorMessage = string.Empty;
+		try
+		{
+			var urlRoute = MarkDownId == null ? "api/markdown/readme/" : $"/api/markdown?currentId={MarkDownId}";
+			var requestMsg = new HttpRequestMessage(HttpMethod.Get, urlRoute);
+			var response = await HttpClient.SendAsync(requestMsg);
+			if (response.IsSuccessStatusCode)
+			{
+				await SetSuccess(response, string.Empty);
+			}
+			else
+			{
+				SetError("an error has occured");
+			}
+		}
+		catch (Exception ex)
+		{
+			SetError(ex.Message);
+		}
+		finally
+		{
+			isLoading = false;
+		}
+	}
+
+	private async Task<string> GetJWT()
+	{
+		var jwt = await jsr.InvokeAsync<string>("localStorage.getItem", "user").ConfigureAwait(false);

```

Comment:
JFI the lib below allow you to have access to local storage  in C#
https://github.com/Blazored/LocalStorage